x/stakeibc/keeper/validator_selection.go

```diff
@@ -244,26 +258,33 @@ func (k Keeper) GetValidatorDelegationDifferences(ctx sdk.Context, hostZone type
 	return delegationDeltas, nil
 }
 
-// This will get the target validator delegation for the given hostZone
-// such that the total validator delegation is equal to the finalDelegation
-// output key is ADDRESS not NAME
-func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone, finalDelegation sdkmath.Int) (map[string]sdkmath.Int, error) {
+// This will split a total delegation amount across validators, according to weights
+// It returns a map of each portion, key'd on validator address
+// Validator's with a slash query in progress are excluded
+func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone, totalDelegation sdkmath.Int) (map[string]sdkmath.Int, error) {
 	// Confirm the expected delegation amount is greater than 0
-	if !finalDelegation.IsPositive() {
+	if !totalDelegation.IsPositive() {
 		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest,
-			"Cannot calculate target delegation if final amount is less than or equal to zero (%v)", finalDelegation)
+			"Cannot calculate target delegation if final amount is less than or equal to zero (%v)", totalDelegation)
+	}
+
+	// Ignore any validators with a slash query in progress
+	validators := []types.Validator{}
+	for _, validator := range hostZone.Validators {
+		if !validator.SlashQueryInProgress {
+			validators = append(validators, *validator)
+		}
 	}
 
 	// Sum the total weight across all validators
-	totalWeight := k.GetTotalValidatorWeight(hostZone)
+	totalWeight := k.GetTotalValidatorWeight(validators)
```

Comment:
Assume we would like to delegate across 5 validators equally, each with 20% weight. But val1 has an ICQs in flight — would `totalWeight` here be 80%, and therefore would each of val2 through val5 receive 25% instead of 20%? 

Thinking about whether we'd want to give each of va2 through val5 20% weight, and keep what would have gone to the first val undelegated, to be delegated next time, OR delegate everything to val2 through val5, some of which would then be redelegated to val1 on the next rebalancing.

Reply:
I must be misunderstanding this logic, based on your back n forth with Ethan in the PR comments. 

To help me undestand, maybe a simple question: in the case I described above, would the delegations made to val2 through val5 be their intended amounts, or slightly more (since val1 is excluded from the denominator when calculating the weights)?

Reply:
> Assume we would like to delegate across 5 validators equally, each with 20% weight. But val1 has an ICQs in flight — would totalWeight here be 80%, and therefore would each of val2 through val5 receive 25% instead of 20%?

yep, that's right, delegations would be slightly more

Reply:
Is your thinking that it's preferable for everything to be delegated and earning rewards but slightly off of the target weights, vs on the target weights but some is undelegated?

Reply:
yeah exactly. I went back and forth on this a bit, but holding back delegations and undelegations feels like a dangerous game. Holding back delegations leads to less rewards. And (maybe this is too dramatic), but holding back undelegations inches slowly towards insolvency

Reply:
Yeah that makes sense, I'm on board with this. Nice use of duration mismatch haha